coreclient/src/contacts/mod.rs

```diff
@@ -78,40 +75,18 @@ impl Contact {
             .get(invited_user_domain)?
             .qs_key_package(self.friendship_token.clone())
             .await?;
+
         let key_package_in = key_package_response.key_package;
+
         // Verify the KeyPackage
         let verified_key_package =
             key_package_in.validate(&RustCrypto::default(), ProtocolVersion::default())?;
-        let pseudonymous_credential = PseudonymousCredential::try_from(
-            verified_key_package.leaf_node().credential().clone(),
-        )?;
-        // Verify the pseudonymous credential
-        let (plaintext, identity_link_key) =
-            pseudonymous_credential.derive_decrypt_and_verify(&self.connection_key)?;
-        // Verify the client credential
-        let incoming_client_credential = StorableClientCredential::verify(
-            &mut *connection,
-            api_clients,
-            plaintext.client_credential,
-        )
-        .await?;
-        // Check that the client credential is the same as the one we have on file.
-        let Some(current_client_credential) = StorableClientCredential::load_by_user_id(
```

Comment:
We should still verify that the `ClientCredential` in the `KeyPackage` is the one we have on file.

Reply:
Uff. We missed it. Thank for point this out. Restored in d0d6c22